sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -404,7 +404,23 @@ namespace Azure { namespace Core {
     {
       if (IsCanceled())
       {
-        throw OperationCanceledException("Request was canceled by context.");
+        throw OperationCanceledException("Request was cancelled by context.");
+      }
+    }
+
+    /**
+     * @brief Throw an exception if the context will be cancelled after the next time \p interval.
+     *
+     * @remark This function can be used to avoid doing an I/O call like sleep for `n` time interval
+     * by checking if the token would be cancelled after that time.
+     *
+     * @param interval The time that is added to now() before checking if token is cancelled.
+     */
+    void ThrowIfWillCancelledAfter(std::chrono::milliseconds interval) const
```

Comment:
Are we going with 1 'l' or with 2?  I see we have `IsCanceled()`, `ThrowIfCanceled()` and now `ThrowIfWillCancelledAfter()`.  

The name of this method is odd?  `ThrowIfWillCancelledAfter()`...  `ThrowIfWillCancelAfter` reads better.

Reply:
with 2.  But updating all the existing occurrences is part of another PR #1256